lib/install/validate/validator.go

```diff
@@ -482,3 +512,68 @@ func (v *Validator) AddDeprecatedFields(ctx context.Context, conf *config.Virtua
 
 	return &dconfig
 }
+
+// helper to grab HostSystem references out of the Session cluster
+func (v *Validator) getHostsInCluster(ctx context.Context) ([]*object.HostSystem, error) {
+	computeHosts, err := v.Session.Cluster.Hosts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// computeHosts shouldn't be zero anymore; if it still is, there's a problem
+	if len(computeHosts) == 0 {
+		return nil, errors.New("Couldn't get list of hosts connected to target cluster")
+	}
+	return computeHosts, nil
+}
+
+// checks if a given datastore ds is writeable by compute resource specified on cmd line --compute-resource
+func (v *Validator) isTargetDatastoreWriteable(ctx context.Context, ds *object.Datastore) bool {
+	defer trace.End(trace.Begin(ds.String()))
+
+	// get compute hosts if we haven't run this method on this object before
+	if len(v.computeHosts) == 0 {
+		var err error
+		v.computeHosts, err = v.getHostsInCluster(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+		}
+	}
+
+	// now get hosts connected to ds
+	attachedHosts, err := ds.AttachedClusterHosts(ctx, v.Session.Cluster)
+	if err != nil {
+		v.NoteIssue(err)
+		return false
+	}
+
+	if len(attachedHosts) == 0 {
+		// this could mean that we're not operating on a cluster..
+		log.Debugf("AttachedClusterHosts returned nothing; searching for unclustered hosts connected to datastore %s", ds.String())
+		attachedHosts, err := ds.AttachedHosts(ctx)
+
+		if err != nil {
+			v.NoteIssue(err)
+			return false
+		}
+
+		if len(attachedHosts) == 0 {
+			// shouldn't be possible to wind up here, but we should get an error message if we do
+			v.NoteIssue(errors.Errorf("No attached hosts found for datastore %s", ds.String()))
+			return false
+		}
+	}
+
+	// find a common host in the two lists
+	for _, c := range v.computeHosts {
+		for _, a := range attachedHosts {
+			if c.Reference() == a.Reference() {
+				return true
```

Comment:
this loop should filter out all hosts not connected to target datastore.
to make sure the v.computeHosts will contains all hosts connected to this datastore.
while the v.computeHosts is empty, your method should return failure, for no hosts connected to all configured datastores.